plugins/modules/site_workflow_manager.py

```diff
@@ -1482,92 +1604,92 @@ def delete_single_site(self, site_id, site_name):
                         break
 
         except Exception as e:
-            self.msg = "Exception occurred while deleting site '{0}' due to: {1}".format(site_name, str(e))
+            self.msg = "Exception occurred while deleting site '{0}' due to: {1}".format(site_name_hierarchy, str(e))
             self.set_operation_result("failed", False, self.msg, "ERROR")
 
         return self
 
-    def delete_floor(self, site_name):
+    def delete_floor(self, site_name_hierarchy):
         """
         Deletes a floor site by ID.
 
         Parameters:
             site_id (str): The ID of the floor site to be deleted.
-            site_name (str): The name of the floor site to be deleted.
+            site_name_hierarchy (str): The name of the floor site to be deleted.
 
         Returns:
             dict: The API response from the 'deletes_a_floor' operation.
         """
         site_id = self.have.get("site_id")
         try:
             self.log(
-                "Deleting floor site: {0} with ID: {1}".format(site_name, site_id), "INFO")
+                "Deleting floor site: {0} with ID: {1}".format(site_name_hierarchy, site_id), "INFO")
             response = self.dnac._exec(
                 family="site_design",
                 function="deletes_a_floor",
                 op_modifies=True,
                 params={'id': site_id},
             )
-            self.log("Successfully deleted floor site: {0}. API response: {1}".format(site_name, response), "DEBUG")
+            self.log("Successfully deleted floor site: {0}. API response: {1}".format(site_name_hierarchy, response), "DEBUG")
             return response
         except Exception as e:
-            self.msg = "Exception occurred while deleting floor site '{0}' with site_id '{1}' due to: {2}".format(site_name, site_id, str(e))
+            self.msg = "Exception occurred while deleting floor site '{0}' with site_id '{1}' due to: {2}".format(site_name_hierarchy, site_id, str(e))
             self.set_operation_result("failed", False, self.msg, "ERROR").check_return_status()
 
-    def delete_building(self, site_name):
+    def delete_building(self, site_name_hierarchy):
         """
         Deletes a building site by ID.
 
         Parameters:
             site_id (str): The ID of the building site to be deleted.
-            site_name (str): The name of the building site to be deleted.
+            site_name_hierarchy (str): The name of the building site to be deleted.
 
         Returns:
             dict: The API response from the 'deletes_a_building' operation.
         """
         site_id = self.have.get("site_id")
```

Comment:
site_id = self.have.get("site_id")
    
    if not site_id:
        self.log("No site ID found for building site: '{}'.".format(site_name_hierarchy), "ERROR")
        return None

Reply:
Added